### ISSUE TITLE
[TF2] Queue gets constantly cancelled without any player input

### ISSUE THREAD

**Issue by h1z1** [ValveSoftware/Source-1-Games#2820]
Hard to describe, it started around the same time the map loading background changed to heavy on a film roll.   In short, 'Start Search' on a Casual match will display the 'Queued for Casual' top message but disappears without any error nor finding any match.  There's nothing helpful in the console either

```
[PartyClient] Requesting queue for 12v12 Casual Match
[PartyClient] Entering queue for match group 12v12 Casual Match
[PartyClient] Leaving queue for match group 12v12 Casual Match
```

It happens randomly and will eventually find a match after restarting the search a few dozen times.

**Comment by 1v** [comment]
Same.

**Comment by josephmate** [comment]
[This post](https://steamcommunity.com/app/440/discussions/0/1698293255134895047/) recommends in console:
```
tf_party_leave
```
This did not work for me.


[This one](https://steamcommunity.com/app/440/discussions/0/2149847423931834060/) recommends
1. restarting game
2. "Verify integrity of game files", potentially multiple times: 
3. restarting computer

None of these worked for me.


Finally, I remember this morning (Sept. 4th 2022), Windows Defender was complaining about win32/hive.zy. When launching TF2 and queueing, I was also seeing the same windows defender popups in the background that I would see when I launched Chrome. [This post](https://answers.microsoft.com/en-us/protect/forum/all/win32hivezy-removal-notification-every-time-i-run/db598180-4b74-4f19-8c1f-117d688caf91) says a windows update is available that should resolve the false positive. After installing the update, I no longer have this issue. Steam uses a custom version of Chromium. Could windows defend have been interfering with TF2 when block the false positive?

Then it again, maybe a sufficient amount of time passed for the problem to resolve itself. 🤷

**Comment by felipsmartins** [comment]
related #4950

**Comment by Gotchfutchian** [comment]
> [This post](https://steamcommunity.com/app/440/discussions/0/1698293255134895047/) recommends in console:
> 
> ```
> tf_party_leave
> ```
> 
> This did not work for me.
> 
> [This one](https://steamcommunity.com/app/440/discussions/0/2149847423931834060/) recommends
> 
>     1. restarting game
> 
>     2. "Verify integrity of game files", potentially multiple times:
> 
>     3. restarting computer
> 
> 
> None of these worked for me.
> 
> Finally, I remember this morning (Sept. 4th 2022), Windows Defender was complaining about win32/hive.zy. When launching TF2 and queueing, I was also seeing the same windows defender popups in the background that I would see when I launched Chrome. [This post](https://answers.microsoft.com/en-us/protect/forum/all/win32hivezy-removal-notification-every-time-i-run/db598180-4b74-4f19-8c1f-117d688caf91) says a windows update is available that should resolve the false positive. After installing the update, I no longer have this issue. Steam uses a custom version of Chromium. Could windows defend have been interfering with TF2 when block the false positive?
> 
> Then it again, maybe a sufficient amount of time passed for the problem to resolve itself. 🤷
> 
> TLDR:
> 
>     1. `tf_party_leave` in console
> 
>     2. restart game
> 
>     3. Verify integrity of game files
> 
>     4. Restart computer
> 
>     5. Check that Windows Defender or your anti virus is not interfering with your game.
> 
>     6. Wait for a while, maybe it will go away on it's own 🤷

That could be an issue, however, this bug also occurs on Linux (clamav not installed)
It is very random when it happens, however.
Even after the update that was released minutes ago, at the time of writing, did not clear me of the bug.

**Comment by AwesomeCoder412412** [comment]
I don't know if I'm screaming into a void here but here's my story:

https://github.com/ValveSoftware/Source-1-Games/issues/4950#issuecomment-1547019503

I have an update. For the better or worse, the wretched glitch has finally returned to my account after  a year of dormancy. My account is consistently displaying it just like in the old days. The time to strike to fix this bug is now. You can look at my posted comment on the other issue for the details, but this glitch is linked to accounts, and it has nothing to do with the hardware or software the tf2 installation resides on.

Here's a link to my account contains the active glitch. Please, fix it. I want to play tf2 again.

https://steamcommunity.com/id/theawesomecoder

**Comment by kisak-valve** [comment]
[TF2] Queue gets constantly cancelled without any player input

Issue transferred from https://github.com/ValveSoftware/Source-1-Games/issues/4950.
@JoriKos posted on 2023-05-09T10:09:45:

Sometimes, the game will start queueing and after a short while it will stop queueing. There is no notification or any indication why it stopped. I have no clue what causes this bug, however, restarting TF2 and/or Steam usually fixes the issue. 

If you are queued up with a party and one person is affected by this bug, it means the whole party can't queue. If said player leaves, the queues will work just fine.

This goes for both Casual and Mann vs. Machine, no idea about competitive.
***
@jh34ghu43gu commented on 2023-05-09T11:36:33:

Sometimes this is because your request to the GC is empty even if your client is showing maps selected. Deselecting everything then reselecting can sometimes fix it if this is the problem. This commonly occurs when people join/leave parties and/or the leader star is passed around. 

Other times it seems to relate to the GC thinking you are in a match you can't leave(?) Which can sometimes be fixed by restarting steam as you said. A lot of months ago this bug was especially egregious and was only fixed when the GC restarted on Tuesday, some of my friends had the bug persist for multiple days until the maintenance freed them; although I haven't heard of anyone running into such drastic effects in some time.
***
@Ashetf2 commented on 2023-05-09T17:52:38:

There should be feedback in the form of a popup to the player when this occurs, mentioning the possible errors. 
***
@Thespikedballofdoom commented on 2023-05-13T19:36:09:

Absolutely, I'm so sick of casting out the person who gets "infected" in the queue.
***
@AwesomeCoder412412 commented on 2023-05-14T22:52:25:

Finally, a place to tell my story.

The year is 2021. I am just getting into TF2, having installed the game in January. I'm greatly enjoying my time with it, and on the fateful day of June 22nd, 2021, it started. Roughly 10 minutes before the tf2 update slated to release that day released, I attempted to queue up for causal with all maps selected. The queue popup was present for roughly 10 seconds before confusingly disappearing the same way it would if I cancel it manually. Confused, I re-queued and got into a match of 2fort after 2 seconds. Shrugging it off, mid-match I saw that there was an update available for tf2. Excitedly, I quit my game to install it. Little did I know what had just started. 

Before I tell the rest of my story about this infuriating glitch, allow me to recount what exactly happens when it occurs.

If an affected account tries to queue for casual, boot camp, mann up, competitive, or any combination of the four, the matchmaking will cancel in 3-20 seconds, varying randomly. It is still possible to get into matches with this glitch, you just have to find a match before the unknown countdown is up.

This glitch persisted on my account for at least 1.5 years, preventing me from enjoying TF2 to the fullest. I must have verified the game files and reinstalled tf2 and steam a million times over the course of that period. I removed all my configs, anything custom. I left and joined countless parties(a proposed solution online for this glitch). I scoured the internet for answers, only finding the occasional Reddit or steam community post all with the same answer: to verify my game files. It never worked. I even moved to a totally different computer, with no transfer of files between them. Nothing worked.

But do you want to know what the worst thing about this godforsaken glitch was? It was contagious.

Over the course of 1.5 years, I tried to introduce 2 of my friends to the game. When I queued up with them, of course, it didn't work. But when they left the party and restarted the game they had the glitch as well. The glitch has been persistent on their accounts for a very long time, and this has prevented them from getting into the game. They don't play anymore because of it, which saddens me. It's possible it still persists on their untouched accounts even today.

After 1.5 years, one day when I depressedly tried queuing for casual, the glitch didn't happen. I was elated. However, the next day it returned. From this point onward it would come and go, with the periods of its absence growing larger and larger. The last definitive date I can give for when it occurred was sometime during Scream Fortress 2022. I can only pray it stays away from my account, but it has a habit of popping up when you least expect it. I wish I learned of the TF2 bug tracker's existence much sooner, as for a large period of time I had an account that was consistently displaying this bug. On the off-chance that something's still there, I'll leave a link to my Steam profile. If my friends still have the glitch happening on their accounts as well I'll update this comment accordingly.

Please TF2 Team, look into this.

https://steamcommunity.com/id/theawesomecoder/
***
@felipsmartins commented on 2023-08-05T23:35:01:

Yeah, that happens to me too. Sadly! I can't queue for Casual Search.
The search stops after some seconds. :(
I've tried:
* uninstall (and remaining files, caches, etc.) steam and TF2;
* Verify game files
* tf_party_leavy command
* empty custom dir
* deleted cfg dir and add `-autoconfig` to lounch options

And nothing worked.

```
[PartyClient] Joining party 614847723575379
[PartyClient] Re-joining party 614847723575379 we previously had queued messages for.  Triggering full resend of options.
[ReliableMsg] PartySetOptions queued for Party ID 614847723575379, Overwrite 1
[ReliableMsg] PartyQueueForMatch successfully sent for Party ID 0, Embedded Options 1
[ReliableMsg] PartySetOptions started for Party ID 614847723575379, Overwrite 1
[ReliableMsg] PartySetOptions successfully sent for Party ID 614847723575379, Overwrite 1
[PartyClient] Leaving queue for match group 12v12 Casual Match
```

`tf_party_debug`

```
] tf_party_debug 
TFParty: ID:00022f3364279853  1 member(s)  LeaderID: [U:1:211083641]
  In 0 queues:
------
party_id: 614847723575379
leader_id: 76561198171349369
member_ids: 76561198171349369
members {
  owns_ticket: true
  competitive_access: true
  player_criteria {
  }
  activity {
    online: true
    client_version: 8227024
  }
}
group_criteria {
  casual_criteria {
    selected_maps_bits: 4290764798
    selected_maps_bits: 1797519214
    selected_maps_bits: 0
    selected_maps_bits: 4196064
    selected_maps_bits: 26
  }
  custom_ping_tolerance: 44
}
leader_ui_state {
  menu_step: k_eTFSyncedMMMenuStep_Configuring_Mode
  match_group: k_eTFMatchGroup_Casual_12v12
}
```

System information (collected from steam client):

```

Computer Information:
Manufacturer: ASUSTeK COMPUTER INC.
Model: ROG STRIX B650-A GAMING WIFI
Form Factor: Desktop
No Touch Input Detected
Processor Information:
CPU Vendor: AuthenticAMD
CPU Brand: AMD Ryzen 9 7900 12-Core Processor
CPU Family: 0x19
CPU Model: 0x61
CPU Stepping: 0x2
CPU Type: 0x0
Speed: 5481 MHz
24 logical processors
12 physical processors
Hyper-threading: Supported
FCMOV: Supported
SSE2: Supported
SSE3: Supported
SSSE3: Supported
SSE4a: Supported
SSE41: Supported
SSE42: Supported
AES: Supported
AVX: Supported
AVX2: Supported
AVX512F: Supported
AVX512PF: Unsupported
AVX512ER: Unsupported
AVX512CD: Supported
AVX512VNNI: Supported
SHA: Supported
CMPXCHG16B: Supported
LAHF/SAHF: Supported
PrefetchW: Unsupported
Operating System Version:
Debian GNU/Linux 12 (bookworm) (64 bit)
Kernel Name: Linux
Kernel Version: 6.1.0-10-amd64
X Server Vendor: The X.Org Foundation
X Server Release: 12101007
X Window Manager: GNOME Shell
Steam Runtime Version: steam-runtime_0.20230606.51628
Video Card:
Driver: NVIDIA Corporation NVIDIA GeForce RTX 3070 Ti/PCIe/SSE2
Driver Version: 4.6.0 NVIDIA 525.125.06
OpenGL Version: 4.6
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 143 Hz
VendorID: 0x10de
DeviceID: 0x2482
Revision Not Detected
Number of Monitors: 3
Number of Logical Video Cards: 1
Primary Display Resolution: 1920 x 1080
Desktop Resolution: 3650 x 2160
Primary Display Size: 20.75" x 11.65" (23.78" diag), 52.7cm x 29.6cm (60.4cm diag)
Primary Bus: PCI Express 16x
Primary VRAM: 8192 MB
Supported MSAA Modes: 2x 4x 8x 16x
Sound card:
Audio device: USB Mixer
Memory:
RAM: 31823 Mb
VR Hardware:
VR Headset: None detected
Miscellaneous:
UI Language: English
LANG: pt_BR.UTF-8
Total Hard Disk Space Available: 936330 MB
Largest Free Hard Disk Block: 793173 MB
Storage:
Number of SSDs: 7
SSD sizes: 4000G,1000G,1000G,1000G,480G,240G,240G
Number of HDDs: 0
Number of removable drives: 0

```
***
@Ruskki commented on 2023-10-08T10:13:39:

It's happening to me as well since a few days now. My friends can't queue with me because of it
***
@NULLYUKI commented on 2023-10-08T15:12:23:

Just noticed that this error probably is already being tracked in #2820 

So this issue is probably a duplicate and it would be better to move all the posted information here into #2820 
***
@AwesomeCoder412412 commented on 2023-10-31T04:34:28:

> Just noticed that this error probably is already being tracked in #2820 
> 
> 
> 
> So this issue is probably a duplicate and it would be better to move all the posted information here into #2820 

I'll provide further updates on #2820 then.
***
@NULLYUKI commented on 2023-10-31T10:25:59:

@kisak-valve Could check again, close this issue and transfer all the provided information into the already existing issue #2820

**Comment by AwesomeCoder412412** [comment]
I forgot to talk about this initially, and have no idea if it helps, but here's the exact moment the glitch returned:

I had successfully queued for all payload maps, gotten into a match of upward, and exited it after finishing the match to completion. I then took a break for an hour, and at roughly 9:35 Los Angeles time I queued for all KOTH maps, and the glitch returned. This iteration of the glitch consistently stops the queue 5 seconds after it starts, instead of varying between 3-20 seconds like back when it persisted on my account for a year. It's actually very possible the glitch returned an exact year after it last returned, as the last sighting before now for me was around this time during scream fortress 2022.

**Comment by felipsmartins** [comment]
whenever it happens to me I have to switch to windows and back to Linux when it solves itself

**Comment by AwesomeCoder412412** [comment]
> whenever it happens to me I have to switch to windows and back to Linux when it solves itself

I've swapped from windows to macOS and macOS to windows when this glitch has occurred and it hasn't done anything. Admittedly, I haven't successfully ran tf2 on Linux yet but I doubt that's the issue. From my experience, this glitch is not tied to the tf2 installation but rather the account. Has switching to windows consistently fixed the glitch for you?

**Comment by AwesomeCoder412412** [comment]
> > Replying to https://github.com/ValveSoftware/Source-1-Games/issues/2820#issuecomment-1236459762
> 
> 
> 
> That could be an issue, however, this bug also occurs on Linux (clamav not installed)
> 
> It is very random when it happens, however.
> 
> Even after the update that was released minutes ago, at the time of writing, did not clear me of the bug.
> 
> 
> 
> Edit: `tf_party_force_update` fixed it this time around for me

'tf_party_force_update' seems like a really promising solution given the ambiguous link between this glitch and the party system, but unfortunately it has not worked at all for me. Considering the amount of solutions to this glitch posted online that seem to only work for the poster and maybe 2-3 others, like tf_party_leave and verifying your game files, perhaps there are multiple glitches with the same/similar symptoms? It would go a long way towards explaining the incredible randomness and inconsistency. 

Some inconsistencies I've noticed that might lend credence to this theory:

1. As detailed above, many different solutions.
2. Some people have had a version of the glitch which is infectious (like myself) and others have been able to play with others in a party who can leave the party without taking the glitch with them.
3. The time this glitch takes seems to vary greatly.

This probably sounds crazy, but at this point I'm going crazy about this glitch. I had finally accepted that it was dead and gone, and it came back to my account to prevent me from playing tf2 once more. I just want it to be over.

**Comment by felipsmartins** [comment]
> Has switching to windows consistently fixed the glitch for you?
Yes! Sometimes it happens to me after a friend of mine joined to my party (but not always). He usually has connections issues. But I don't know if my friend is the culprit this glitch or not.

**Comment by NULLYUKI** [comment]
After rereading through the issue and thinking about the possible reason, I suspect that issue is being caused by a network error maybe.

**Comment by SlenLen** [comment]
I too experience this issue. The menu says it's queued for casual for ~30 seconds before just stopping without any notification.
This makes it impossible to play casual. Community servers work fine.
None of the suggested fixes have worked for me.
I tried:
- restarting game
- selecting and deselecting game modes
- verifying game files
- reinstalling the game
- running `tf_party_leave` in the console
- deleting my config

This issue does not occur on Windows, but it still persists even after successfuly finding a match and then switching to Linux again.
So far I have not been able to fix this bug after it started appearing about 3 days ago.

**Comment by 1v** [comment]
@SlenLen other workaround is to create server.

**Comment by SlenLen** [comment]
@1v That workaround also didn't fix it for me.
I tried queueing while still on the server and after I left it, with and without running `tf_party_leave` inbetween, but queueing still results in nothing but that bug.
I think the command doesn't change anything because I get put into the bugged party (assuming it even is an issue of the party) regardless, as suggested by the command output `[PartyClient] Re-joining party 625632137069922 we previously had queued messages for.  Triggering full resend of options.`

**Comment by NULLYUKI** [comment]
@SlenLen 
> I think the command doesn't change anything because I get put into the bugged party (assuming it even is an issue of the party) regardless, as suggested by the command output `[PartyClient] Re-joining party 625632137069922 we previously had queued messages for.  Triggering full resend of options.`

If it might be a issue with the party system, does it maybe has to do something with the party settings from "Team Fortress 2" (If there are any not sure right now, need to check that) or with the privacy settings from Steam?

**Comment by SlenLen** [comment]
@NULLYUKI As far as I'm aware there are no party settings in Team Fortress 2. There are matchmaking settings, however they don't seem to affect anything regarding the queueing issue.
Steam's privacy settings make no difference as well. I tried with my profile set to "private" and with my profile set to public along with my game details etc. yet nothing changed.

**Comment by jh34ghu43gu** [comment]
@SlenLen I haven't been able to try this cause I haven't gotten this bug in months but joining someone else's party and leaving might be a potential option.

You might also have the turbo-locked version of this bug I mentioned that only gets fixed on the Tuesday restart (my original reply is in Kisak's reply).

**Comment by SlenLen** [comment]
@jh34ghu43gu Joining someone else's party doesn't fix the bug either. The party wasn't able to find a match for as long as I was in it. When I left, everybody else in the party could find a match withouth any issues, but the bug persisted for me.

The server resart on tuesday fixed it however and now I don't have any problems with matchmaking anymore (for now at least).

**Comment by constantitus** [comment]
Experiencing the issue right now on linux. None of the aforementioned solutions work. I can play from another account on the same machine. Installing the game on another linux machine and playing from the same account still has the issue, so it's definitely an account-related issue. I haven't tried it on windows, nor do I intend to.

**Comment by ItsNeverLight** [comment]
Still having this issue... 
`[PartyClient] No longer leader of party 638945105169651
[PartyClient] Entering queue for match group 12v12 Casual Match
[PartyClient] Leaving queue for match group 12v12 Casual Match
[PartyClient] Entering queue for match group 12v12 Casual Match
[PartyClient] Leaving queue for match group 12v12 Casual Match
`

**Comment by JoriKos** [comment]
It's not necessary to constantly mention this issue is still happening. See the [readme](https://github.com/ValveSoftware/Source-1-Games):

`Do not repeatedly update an open issue remarking that the issue persists.`

**Comment by Nephyrin** [comment]
If you are encountering this issue, the following would be helpful to know:
- OS
  - If using Linux, distro and Steam install method (e.g. flatpak)
- Any other steam mods/customization that may be active
- If the problem started at any identifiable moment, such as Tuesday weekly Steam maintenance, a specific update, etc..
- If rebooting solves the issue
- If opting into or out of the Steam Client Beta solves the issue
- If `tf_party_leave` solves the issue

Additionally, it would be helpful to create a Steam Support ticket which would help us collect more debugging information from you.

**Comment by JoriKos** [comment]
Whenever I experienced it I

* Was using Windows 10
* Had custom HUD and hitsound/killsound enabled
* Don't recall anything specific, happening outside of people joining the party.
* Rebooting can sometimes help resolve the issues. When in a party with multiple people it always seems to be only one person that causes it (e.g. if they leave the party, we can queue fine)
* Haven't used the Steam beta

If it ever happens again I will try and make a Support ticket and try `tf_party_leave`. Is there anything specific I need to include in the Support ticket?

**Comment by SoildSnek** [comment]
In response to Nephyrin's comment

OS: Manjaro 6.9.12-3
Mods: None

Issue started today Tuesday 8/20/24 post steam maintenance window.

I have tried every solution listed in this thread so far with nothing fixing it.

Opting in and tested with the issue persisting 

Opted back out still nothing

attempted tf_party_leave and tf_party_force_update

Will open a steam ticket for this as well.

**Comment by SoildSnek** [comment]
Update

This issue is completely fixed for me after putting in a steam support ticket. I linked this git-hub issue as well as my post above. 

It seems that the cause of the issue happened:

- On a steam maintenance day (Tuesday)
- After party leader was changed

I threw in a support ticket through steam and they fixed it on the back end within the day. This seems to be the best solution that works so far.

**Comment by JayTeeMacquarie** [comment]
Still an issue.

**Comment by Midlou** [comment]
I can confirm that this problem is still happening

**Comment by constantitus** [comment]
Instead of replying to the issue without providing any meaningful information, go file a support ticket.
In my case, the issue didn't last more than a week. I did file a support ticket, but the issue disappeared before they replied.

**Comment by dbt0815** [comment]
I filed a support ticket. This shit needs to stop.

**Comment by LockBitSupp** [comment]
I can confirm this is still an active issue, I've been unable to queue in a party for coming on two years now, none of the suggested mitigations in related github issues have helped at all, as a result I've almost entirely quit playing TF2. Fix your shit valve, this is getting ridiculous.

**Comment by Nephyrin** [comment]
> I can confirm this is still an active issue, I've been unable to queue in a party for coming on two years now, none of the suggested mitigations in related github issues have helped at all, as a result I've almost entirely quit playing TF2. Fix your shit valve, this is getting ridiculous.

Please file a support ticket and mention this issue ( `https://github.com/ValveSoftware/Source-1-Games/issues/2820` ) so we can investigate.

**Comment by LockBitSupp** [comment]
> > I can confirm this is still an active issue, I've been unable to queue in a party for coming on two years now, none of the suggested mitigations in related github issues have helped at all, as a result I've almost entirely quit playing TF2. Fix your shit valve, this is getting ridiculous.
> 
> Please file a support ticket and mention this issue ( `https://github.com/ValveSoftware/Source-1-Games/issues/2820` ) so we can investigate.

already have, awaiting reply.

**Comment by gaelcoral** [comment]
![image](https://github.com/user-attachments/assets/6cceb73b-6da1-4992-8ae9-7103713d8c19)

**Comment by AwesomeCoder412412** [comment]
> ![image](https://github.com/user-attachments/assets/6cceb73b-6da1-4992-8ae9-7103713d8c19)
> 
> 

Maybe... just maybe... they finally addressed it?

**Comment by Thespikedballofdoom** [comment]
Liberation... Let's give it a month and see if anyone gets hit by this. If they do it'll at least have an error now

**Comment by gaelcoral** [comment]
> Liberation... Let's give it a month and see if anyone gets hit by this. If they do it'll at least have an error now

![image](https://github.com/user-attachments/assets/d6c0b4a5-d072-4c58-a5d5-ae791da5285b)

**Comment by Nephyrin** [comment]
Marking this as closed, there have been a number of backend fixes to matchmaking, and most queue ejections should now result in a salient error.
